File: app/src/main/java/com/example/instabug/fcm/InstabugInstanceIDListenerService.java
package com.example.instabug.fcm;

import androidx.annotation.NonNull;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.instabug.chat.Replies;

/**
 * Created by vezikon on 6/27/16.
 */
public class InstabugInstanceIDListenerService extends FirebaseMessagingService {

    /**
     * Called when a new token for the default Firebase project is generated.
     * This is invoked after app install when a token is first generated, and again if the token changes.
     *
     * @param token – The token used for sending messages to this application instance. This token is the same as the one retrieved by FirebaseMessaging.getToken().
     */
    @Override
    public void onNewToken(@NonNull String token) {
        super.onNewToken(token);
        Replies.setPushNotificationRegistrationToken(token);
    }
}

File: app/src/main/java/com/example/instabug/fcm/InstabugInstanceIDListenerService.kt
package com.example.instabug.fcm

import android.content.Intent

import com.google.firebase.iid.FirebaseInstanceIdService

/**
 * Created by vezikon on 6/27/16.
 */
class InstabugInstanceIDListenerService : FirebaseInstanceIdService() {

    /**
     * Called if InstanceID token is updated. This may occur if the security of
     * the previous token had been compromised. This call is initiated by the
     * InstanceID provider.
     */
    // [START refresh_token]
    override fun onTokenRefresh() {
        // Fetch updated Instance ID token and notify our app's server of any changes (if applicable).
        val intent = Intent(this, RegistrationIntentService::class.java)
        startService(intent)
    }

    companion object {

        private const val TAG = "InstabugInstanceIDLS"
    }
    // [END refresh_token]
}